src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoaderStep.java

```diff
@@ -15,68 +15,116 @@
+import org.springframework.batch.item.file.FlatFileParseException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
-import org.springframework.stereotype.Component;
+import org.springframework.data.mongodb.core.MongoOperations;
+
+import uk.ac.ebi.eva.commons.models.data.Variant;
 import uk.ac.ebi.eva.pipeline.configuration.JobOptions;
 import uk.ac.ebi.eva.pipeline.configuration.JobParametersNames;
-import uk.ac.ebi.eva.utils.FileUtils;
-import uk.ac.ebi.eva.utils.URLHelper;
-
-import java.net.URI;
-import java.nio.file.Path;
-import java.nio.file.Paths;
+import uk.ac.ebi.eva.pipeline.io.readers.AggregatedVcfReader;
+import uk.ac.ebi.eva.pipeline.io.readers.UnwindingItemStreamReader;
+import uk.ac.ebi.eva.pipeline.io.readers.VcfReader;
+import uk.ac.ebi.eva.pipeline.io.writers.VariantMongoWriter;
+import uk.ac.ebi.eva.pipeline.listeners.SkippedItemListener;
+import uk.ac.ebi.eva.pipeline.model.converters.data.VariantToMongoDbObjectConverter;
+import uk.ac.ebi.eva.utils.MongoDBHelper;
 
 /**
- * Tasklet that loads transformed variants into mongoDB
+ * Step that loads transformed variants into mongoDB
  * <p>
  * Input: transformed variants file (variants.json.gz)
  * Output: variants loaded into mongodb
  */
-@Component
-@StepScope
-@Import({JobOptions.class})
-public class VariantLoaderStep implements Tasklet {
-    private static final Logger logger = LoggerFactory.getLogger(VariantLoaderStep.class);
+@Configuration
+@EnableBatchProcessing
+@Import(JobOptions.class)
+public class VariantLoaderStep {
+    public static final String LOAD_VARIANTS = "Load variants";
 
     @Autowired
-    private JobOptions jobOptions;
+    private StepBuilderFactory stepBuilderFactory;
 
-    @Override
-    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
-        ObjectMap variantOptions = jobOptions.getVariantOptions();
-        ObjectMap pipelineOptions = jobOptions.getPipelineOptions();
+    @Autowired
+    private JobOptions jobOptions;
 
-        VariantStorageManager variantStorageManager = StorageManagerFactory.getVariantStorageManager();// TODO add mongo
+    @Autowired
+    private UnwindingItemStreamReader<Variant> reader;
 
-        URI outdirUri = FileUtils.getPathUri(pipelineOptions.getString(JobParametersNames.OUTPUT_DIR),true);
-        URI nextFileUri = URLHelper.createUri(pipelineOptions.getString(JobParametersNames.INPUT_VCF));
+    @Autowired
+    private VariantMongoWriter variantMongoWriter;
+    
+    @Bean
+    @Qualifier("variantsLoadStep")
+    public Step variantsLoadStep() throws Exception {
+        return stepBuilderFactory.get(LOAD_VARIANTS).<Variant, Variant>chunk(10)
```

Comment:
From now on, I will put a "hooray" reaction in a review comment to mark it as already done, and an update of the PR pushing more commits (containing that change) can be expected soon.